bf.c

```diff
@@ -0,0 +1,107 @@
+#include <err.h>
+#include <fcntl.h>
+#include <stdbool.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <sys/stat.h>
+#include <unistd.h>
+
+#define CELLS_LEN 30000
+
+void interp(char *prog, char *prog_end, char *cells, char *cells_end) {
+    char *instr = prog;
+    char *cell = cells;
+    while (instr < prog_end) {
+        switch (*instr) {
+            case '>': {
+                if (cell++ == cells_end)
+                    errx(1, "out of memory");
+                instr++;
+                break;
+            }
+            case '<': {
+                if (cell > cells)
+                    cell--;
+                instr++;
+                break;
+            }
+            case '+': {
+                (*cell)++;
+                instr++;
+                break;
+            }
+            case '-': {
+                (*cell)--;
+                instr++;
+                break;
+            }
+            case '.': {
+                putchar(*cell);
+                instr++;
+                break;
+            }
+            case ',': {
+                if (read(STDIN_FILENO, cell, 1) == -1)
```

Comment:
I thought you were not a fan of unbraced conditionals?

Reply:
When in Rome, dear boy.

Reply:
You've changed :P